include/usb_cam/usb_cam.hpp

```diff
@@ -288,37 +293,57 @@ class UsbCam
     using usb_cam::formats::MJPEG2RGB;
     using usb_cam::formats::M4202RGB;
 
-    if (str == "rgb8") {
+    if (pixFmtStr == "rgb8") {
       m_image.pixel_format = std::make_shared<RGB8>();
-    } else if (str == "yuyv") {
+    } else if (pixFmtStr == "yuyv") {
       m_image.pixel_format = std::make_shared<YUYV>();
-    } else if (str == "yuyv2rgb") {
+    } else if (pixFmtStr == "yuyv2rgb") {
       // number of pixels required for conversion method
       m_image.pixel_format = std::make_shared<YUYV2RGB>(m_image.number_of_pixels);
-    } else if (str == "uyvy") {
+    } else if (pixFmtStr == "uyvy") {
       m_image.pixel_format = std::make_shared<UYVY>();
-    } else if (str == "uyvy2rgb") {
+    } else if (pixFmtStr == "uyvy2rgb") {
       // number of pixels required for conversion method
       m_image.pixel_format = std::make_shared<UYVY2RGB>(m_image.number_of_pixels);
-    } else if (str == "mjpeg2rgb") {
+    } else if (pixFmtStr == "mjpeg2rgb") {
       m_image.pixel_format = std::make_shared<MJPEG2RGB>(
-        m_image.width, m_image.height);
-    } else if (str == "m4202rgb") {
+        m_image.width, m_image.height, get_av_pixel_format_from_string(avDevFmtStr));
+    } else if (pixFmtStr == "m4202rgb") {
       m_image.pixel_format = std::make_shared<M4202RGB>(
         m_image.width, m_image.height);
-    } else if (str == "mono8") {
+    } else if (pixFmtStr == "mono8") {
       m_image.pixel_format = std::make_shared<MONO8>();
-    } else if (str == "mono16") {
+    } else if (pixFmtStr == "mono16") {
       m_image.pixel_format = std::make_shared<MONO16>();
-    } else if (str == "y102mono8") {
+    } else if (pixFmtStr == "y102mono8") {
       m_image.pixel_format = std::make_shared<Y102MONO8>(m_image.number_of_pixels);
     } else {
-      throw std::invalid_argument("Unsupported pixel format specified: " + str);
+      throw std::invalid_argument("Unsupported pixel format specified: " + pixFmtStr);
     }
 
     return m_image.pixel_format;
   }
 
+  inline AVPixelFormat get_av_pixel_format_from_string (const std::string& str)
```

Comment:
@boitumeloruf ideally we keep the AV* ffmpeg stuff out of this file if we can. Can we move this function to somewhere else?

Reply:
Well, we could move this function to the `av_picel_format_herlper.hpp` header file. In this way, all the libavcodec specific code is in one helper file.

Reply:
@boitumeloruf I think thats fine and put that new file within the `formats` directory 👍🏼